test/unit/grant-types/password-grant-type.spec.ts

```diff
@@ -0,0 +1,78 @@
+import * as should from 'should';
+import * as sinon from 'sinon';
+import { PasswordGrantType } from '../../../lib/grant-types';
+import { Request } from '../../../lib/request';
+
+/**
+ * Test `PasswordGrantType`.
+ */
+
+describe('PasswordGrantType', () => {
+  describe('getUser()', () => {
+    it('should call `model.getUser()`', async () => {
+      const model = {
+        getUser: sinon.stub().returns(true),
+        saveToken() {},
+      };
+      const handler = new PasswordGrantType({
+        accessTokenLifetime: 120,
+        model,
+      });
+      const request = new Request({
+        body: { username: 'foo', password: 'bar' },
+        headers: {},
+        method: 'ANY',
+        query: {},
+      });
+      try {
+        await handler.getUser(request);
+        model.getUser.callCount.should.equal(1);
+        model.getUser.firstCall.args.should.have.length(2);
+        model.getUser.firstCall.args[0].should.equal('foo');
+        model.getUser.firstCall.args[1].should.equal('bar');
+        model.getUser.firstCall.thisValue.should.equal(model);
+      } catch (error) {
+        should.fail('should.fail', '');
+      }
```

Comment:
This conversion to async/await now makes it obvious that the original tests were problematic: A failure in the `model.getUser.` shoulds will be caught by the catch, which then drops all and every detail about the actual problem.

This should be written as:
```typescript
      try {
        await handler.getUser(request);
      } catch (error) {
        should.fail('should.fail', '');
      }
        model.getUser.callCount.should.equal(1);
        model.getUser.firstCall.args.should.have.length(2);
        model.getUser.firstCall.args[0].should.equal('foo');
        model.getUser.firstCall.args[1].should.equal('bar');
        model.getUser.firstCall.thisValue.should.equal(model);
```
At this point then the question appears what the catch actually helps with, apart from equally just killing contextual information. I think best would be to let mocha handle the problem itself:
```typescript
        await handler.getUser(request);
        model.getUser.callCount.should.equal(1);
        model.getUser.firstCall.args.should.have.length(2);
        model.getUser.firstCall.args[0].should.equal('foo');
        model.getUser.firstCall.args[1].should.equal('bar');
        model.getUser.firstCall.thisValue.should.equal(model);
```

In the original code using explicit promises this would be equivalent to dropping the `catch`.

(Same problem appears in many tests)

Reply:
`should.fail` was expected. 
with your change, it will be changed to expect a `success`.